### PR TITLE
Refactor `util::prefs` operations to be methods on static struct.

### PR DESCRIPTION
**Pull request by frewsxcv**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because _____

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12178)

### CODE REVIEW

**Comment by frewsxcv** [comment]
I was never a fan of the assortment of functions in `components/util/pref.rs`. I publicized the `static` datastructure in the module and added some methods around it. I think it's cleaner now, but I don't feel strongly. Just experimenting here.

**Comment by ConnorGBrewster** [comment]
Tidy:
```
Checking files for tidiness...
./components/compositing/compositor.rs:53: use statement is not in alphabetical order
	expected: util::print_tree::PrintTree
	found: util::opts
./components/servo/lib.rs:85: use statement is not in alphabetical order
	expected: util::resource_files::resources_dir_path
	found: util::opts
```

File: components/util/prefs.rs [hunk 1]
```diff
@@ -14,9 +14,9 @@ use std::path::PathBuf;
 use std::sync::{Arc, Mutex};
 
 lazy_static! {
-    static ref PREFS: Arc<Mutex<HashMap<String, Pref>>> = {
+    pub static ref PREFS: Preferences = {
         let prefs = read_prefs().unwrap_or(HashMap::new());
```
Comment (emilio) [inline components/util/prefs.rs]: now you're at it, could you change this `unwrap_or(HashMap::new())` for `unwrap_or_else(HashMap::new)`?

It's a nit though, so feel free to not do it :)

File: components/util/prefs.rs [hunk 2]
```diff
@@ -14,9 +14,9 @@ use std::path::PathBuf;
...
-        Arc::new(Mutex::new(prefs))
+        Preferences(Arc::new(Mutex::new(prefs)))
```
Comment (emilio) [inline components/util/prefs.rs]: Actually, I'm wondering if we could use a `RWLock` instead of a `Mutex`, this should avoid locking the world when reading preferences, which is the common case.

Comment (frewsxcv) [inline components/util/prefs.rs]: Valid suggestion, though could we make that a separate PR?

File: components/util/prefs.rs (hunk 1 above)
Comment (frewsxcv) [inline components/util/prefs.rs]: https://github.com/servo/servo/pull/12178/commits/6bc0f58a482cece38a158a2504e141e73c46a2fc

File: components/util/prefs.rs (hunk 2 above)
Comment (emilio) [inline components/util/prefs.rs]: Sure :)

**Comment by emilio** [comment]
maybe `.ok().unwrap_or_else(HashMap::new)` is a bit cleaner? Not too much though, so feel free to r=me with or without that.

@bors-servo: delegate+
